src/MyQConnector/Connectors/MyQConnector.cs

```diff
@@ -0,0 +1,199 @@
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Linq;
+
+namespace Innovatrics.SmartFace.Integrations.MyQConnectorNamespace.Connectors
+{
+    public class MyQConnector : IMyQConnector
+    {
+        private readonly ILogger logger;
+        private readonly IConfiguration configuration;
+        private readonly IHttpClientFactory httpClientFactory;
+
+        private string clientId;
+        private string clientSecret;
+        private string scope;
+        private int loginInfoType;
+        private string MyQHostname;
+        private int MyQPort;
+        private string SmartFaceURL;
+        private bool bypassSslValidation;
+
+        public MyQConnector(ILogger logger, IConfiguration configuration, IHttpClientFactory httpClientFactory)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            this.httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+
+            clientId = configuration.GetValue<string>("MyQConfiguration:clientId") ?? throw new InvalidOperationException("clientId is required");
+            clientSecret = configuration.GetValue<string>("MyQConfiguration:clientSecret") ?? throw new InvalidOperationException("clientSecret is required");
+            scope = configuration.GetValue<string>("MyQConfiguration:scope") ?? throw new InvalidOperationException("scope is required");
+            loginInfoType = configuration.GetValue<int>("MyQConfiguration:loginInfoType");
+            MyQHostname = configuration.GetValue<string>("MyQConfiguration:MyQHostname") ?? throw new InvalidOperationException("MyQHostname is required");
+            MyQPort = configuration.GetValue<int>("MyQConfiguration:MyQPort");
+            SmartFaceURL = configuration.GetValue<string>("MyQConfiguration:SmartFaceURL");
+            bypassSslValidation = configuration.GetValue<bool>("MyQConfiguration:BypassSslValidation");
+        }
+
+        public async Task OpenAsync(string myqPrinter, Guid myqStreamId, string watchlistMemberId)
+        {
+            this.logger.Information("MyQ Printer Unlocking");
+            this.logger.Information($"WatchlistMemberID: {watchlistMemberId}");
+
+            try
+            {
+                string email = await GetEmailFromSmartFaceAPI(watchlistMemberId);
+                string token = await AuthenticateWithMyQAPI();
+                string userInfo = await GetUserInfo(email, token);
+                string userToken = await AuthenticateUserWithMyQAPI(userInfo);
+                await UnlockPrinter(myqPrinter, myqStreamId, userToken);
+            }
+            catch (Exception ex)
+            {
+                this.logger.Error(ex, "Error occurred in OpenAsync");
```

Comment:
co sa potom deje s vynimkou? Ak open neprejde, tak aktualne to ticho zlyha, ta exception sa nepropaguje dalej. Malo by to byt:

catch (Exception ex)
            {
                this.logger.Error(ex, "Error occurred in OpenAsync"); 
                throw;
            }

Reply:
upravene fixnute